app/services/CreditHistoryService.scala

```diff
@@ -74,20 +75,36 @@ class CreditHistoryService @Inject()(incomeTaxViewChangeConnector: IncomeTaxView
 
   def getCreditsHistory(calendarYear: Int, nino: String)
                        (implicit hc: HeaderCarrier, user: MtdItUser[_]): Future[Either[CreditHistoryError.type, List[CreditDetailModel]]] = {
+
     for {
       creditModelForTaxYear <- getCreditsByTaxYear(calendarYear, nino)
       creditModelForTaxYearPlusOne <- getCreditsByTaxYear(calendarYear + 1, nino)
     } yield (creditModelForTaxYear, creditModelForTaxYearPlusOne) match {
       case (Right(creditModelTY), Right(creditModelTYandOne)) =>
-        Right((creditModelTY ++ creditModelTYandOne).filter(creditDetailModel => getTaxYearAsInt(creditDetailModel.documentDetail.taxYear) == calendarYear))
+        val creditsForTaxYearAndPlusOne =
+          (creditModelTY ++ creditModelTYandOne).filter(creditDetailModel => getTaxYearAsInt(creditDetailModel.documentDetail.taxYear) == calendarYear)
+        Right(filterExcludedCredits(creditsForTaxYearAndPlusOne))
       case (Right(creditModelTY), Left(_)) =>
-        Right(creditModelTY.filter(creditDetailModel => getTaxYearAsInt(creditDetailModel.documentDetail.taxYear) == calendarYear))
+        val creditsForTaxYear =
+          creditModelTY.filter(creditDetailModel => getTaxYearAsInt(creditDetailModel.documentDetail.taxYear) == calendarYear)
+        Right(filterExcludedCredits(creditsForTaxYear))
       case (Left(_), Right(creditModelTYandOne)) =>
-        Right(creditModelTYandOne.filter(creditDetailModel => getTaxYearAsInt(creditDetailModel.documentDetail.taxYear) == calendarYear))
+        val creditsForTaxYearPlusOne =
+          creditModelTYandOne.filter(creditDetailModel => getTaxYearAsInt(creditDetailModel.documentDetail.taxYear) == calendarYear)
+        Right(filterExcludedCredits(creditsForTaxYearPlusOne))
       case (_, _) =>
         Left(CreditHistoryError)
     }
   }
+
+  private def filterExcludedCredits(credits: List[CreditDetailModel]): List[CreditDetailModel] = {
```

Comment:
in general, we would prefer all FS related check to remain on the controller level for the consistency; but we can do this refactoring in the next round of changes;